remapping_pipeline.nf

```diff
@@ -0,0 +1,350 @@
+
+    # Replace the colon separators with "|":
+    # Storing this information for later on in the script when we split the name by "|" to extract the relevant 
+    # information (ALT allele, QUAL, FILT, INFO)
+    # This is done because the INFO column can contain ":", which means we wouldn't be able to split by ":", so "|" was 
+    # chosen
+    sed -i 's/:/|/' variants_reads.fa
+    '''
+}
+
+/*
+ * Extract information about the original variants and put it in the fasta header
+ */
+process extractVariantInfoToFastaHeader {
+
+    input:  
+        path "flanking.bed" from flanking_bed
+        path "variants_reads.fa" from variants_reads
+    
+    output:
+        path "variant_reads.out.fa" into variant_reads_with_info
+
+    // Disable the string interpolation using single quotes
+    // https://www.nextflow.io/docs/latest/script.html#string-interpolation
+    '''
+    # Store ref bases
+    cut -f 6 flanking.bed > old_ref_bases.txt
+
+    # Store rsIDs
+    cut -f 4 flanking.bed > rsIDs.txt
+
+    # Store variant bases
+    cut -f 7  flanking.bed > variant_bases.txt
+
+    # Store the other vcf columns
+    cut -f 5,8,9 flanking.bed > qual_filt_info.txt
+
+    # Paste the names, variant bases, then fasta sequences into a new file
+    paste <(grep '^>' variants_reads.fa) old_ref_bases.txt variant_bases.txt rsIDs.txt qual_filt_info.txt \
+    <(grep -v '^>' variants_reads.fa) > temp.txt
+
+    # Reformat the fasta ID: inconsistencies in the separators, and no new line before the sequence mean that this next 
+    # command is a bit ugly
+    # Input:
+    # >[chr]|[pos interval]   [REF]       [ALT]       [rsID]    [QUAL] [FILT] [INFO]      [seq]
+    # Replace all spaces and tabs with "|":
+    # >[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO]|[seq]
+    # Replace the last "|" with a space (this is between the header and the sequence):
+    # >[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO] [seq]
+    # And finally replace the space with a newline:
+    # Output:
+    # >[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO]
+    # [seq]
+    sed 's/\\t/|/g; s/ /|/g; s/\\(.*\\)|/\\1 /' temp.txt | tr ' ' '\\n' > variant_reads.out.fa
+    '''
+}
+
+/*
+ * Align sequence with bowtie2
+ */
+process alignWithBowtie {
+
+    // Memory required is 5 times the size of the fasta in Bytes or at least 1GB
+    memory  Math.max(file(params.newgenome).size() * 5, 1073741824) + ' B'
+
+    input:  
+        path "variant_reads.fa" from variant_reads_with_info
+        // This will get the directory containing the bowtie index linked in the directory
+        file 'bowtie_index' from newgenome_dir
+
+
+    output:
+        path "reads_aligned.bam" into reads_aligned_bam
+
+    """
+    bowtie2 -k 2 --np 0 -f -x bowtie_index/${newgenome_basename} variant_reads.fa | samtools view -bS - > reads_aligned.bam
+    """
+}
+
+/*
+ * Sort the bam file with samtools
```

Comment:
```suggestion
 * Sort the bam file with samtools and index the result.
```

Reply:
@tcezard And also here. Maybe a batch of accepted suggestions didn't get through? GitHub sometimes does those things